src/vm/eepolicy.cpp

```diff
@@ -1105,6 +1105,42 @@ void EEPolicy::HandleExitProcess(ShutdownCompleteAction sca)
     HandleExitProcessHelper(action, 0, sca);
 }
 
+//---------------------------------------------------------------------------------------
+//
+// Writes a log error message to stderr of an unhandled exception.
+//
+// Arguments:
+//    pExceptionInfo - Exception information
+//
+// Return Value:
+//    None
+//
+void DoLogForUnhandledException(LPCWSTR pszMessage, PEXCEPTION_POINTERS pExceptionInfo)
+{
+    WRAPPER_NO_CONTRACT;
+
+    EX_TRY
+    {
+        StackSString s;
+        InlineSString<80> ssErrorFormat;
+        if(!ssErrorFormat.LoadResource(CCompRC::Optional, IDS_ER_UNHANDLEDEXCEPTIONINFO))
+            ssErrorFormat.Set(W("exception code %1, exception address %2"));
+        SmallStackSString exceptionCodeString;
+        exceptionCodeString.Printf(W("%x"), pExceptionInfo->ExceptionRecord->ExceptionCode);
+        SmallStackSString addressString;
+        addressString.Printf(W("%p"), (UINT_PTR)pExceptionInfo->ExceptionRecord->ExceptionAddress);
+        s.FormatMessage(FORMAT_MESSAGE_FROM_STRING, (LPCWSTR)ssErrorFormat, 0, 0, exceptionCodeString, addressString);
+        LPCWSTR stackMsg = s.GetUnicode();
+
+        PrintToStdErrW((WCHAR*)pszMessage);
```

Comment:
I think the printing of `pszMessage` should be done as the very first thing. The system is potentially in unstable state when this method is called, and so the message formatting can crash. It would be nice to print at least something before the crash.